mllib/src/main/scala/org/apache/spark/ml/r/AFTSurvivalRegressionWrapper.scala

```diff
@@ -129,7 +129,9 @@ private[r] object AFTSurvivalRegressionWrapper extends MLReadable[AFTSurvivalReg
       val rMetadata = ("class" -> instance.getClass.getName) ~
         ("features" -> instance.features.toImmutableArraySeq)
       val rMetadataJson: String = compact(render(rMetadata))
-      sc.parallelize(Seq(rMetadataJson), 1).saveAsTextFile(rMetadataPath)
+      sparkSession.createDataFrame(
+        Seq(Tuple1(rMetadataJson))
+      ).repartition(1).write.text(rMetadataPath)
```

Comment:
Can we avoid repartition?

Reply:
No, the single partition is needed to avoid creating empty files, even though there is only one row:

```
spark.range(1).select(sf.lit("123")).write.text("/tmp/test-test")
```

```
(base) ➜  runtime git:(master) ls /tmp/test-test
_SUCCESS                                                 part-00011-e548682e-21a5-41ea-93bc-0089efccbf4f-c000.txt
part-00000-e548682e-21a5-41ea-93bc-0089efccbf4f-c000.txt
(base) ➜  runtime git:(master) wc -l /tmp/test-test/*
       0 /tmp/test-test/_SUCCESS
       0 /tmp/test-test/part-00000-e548682e-21a5-41ea-93bc-0089efccbf4f-c000.txt
       1 /tmp/test-test/part-00011-e548682e-21a5-41ea-93bc-0089efccbf4f-c000.txt
       1 total
```

I think we can use `coalesce(1)` to avoid this small shuffle

Reply:
This is single row, and will produce one partition, see `LocalTableScanExec`.

```scala
spark.createDataFrame(Seq(Tuple1("a"))).write.text("/tmp/abc")
```

write single file.

Reply:
Made a followup: https://github.com/apache/spark/pull/47341

Reply:
ah, I see, this `LocalTableScanExec` ganrantees single row -> single partition.

It will be better if we can specify the `numPartitions` somewhere

Reply:
Yeah I think so but removing it should work for now.